api/app/auto_spatial_advisory/tpi_area_summary.py

```diff
@@ -0,0 +1,45 @@
+from datetime import date
+import numpy as np
+from app.auto_spatial_advisory.run_type import RunType
+from app.utils.s3 import get_client
+from app import config
+from app.utils.geospatial import GeospatialOptions, Vector2RasterOptions, read_raster_into_memory
+
+
+async def summarize_advisory_area_by_tpi(forecast_or_actual: RunType, for_date: date):
+    """
+    Masks classified HFI against classified TPI and counts frequency of TPI classes with classified HFI pixels.
+
+    :return: Mapping of TPI class to HFI pixel count
+    """
+    pmtiles_filename = f'hfi{for_date.strftime("%Y%m%d")}.pmtiles'
+    iso_date = for_date.isoformat()
+
+    async with get_client() as (client, bucket):
+        tpi_result = await read_raster_into_memory(
+            client,
+            bucket,
+            f'dem/tpi/{config.get("CLASSIFIED_TPI_DEM_NAME")}',
+            GeospatialOptions(include_geotransform=True, include_projection=True, include_x_size=True, include_y_size=True),
+        )
+        hfi_result = await read_raster_into_memory(
```

Comment:
It looks like we're massaging the HFI layer to match the geotransform, projection and dimensions of the tpi layer? We might be able to avoid this extra processing if we massage the tpi layer to match the HFI layer when we initially create the tpi layer. This could help to preserve the 'original` HFI pixel values which we loose when we do the reprojection. We could check in with Brett on Monday to see if my thoughts make sense.

Reply:
> It looks like we're massaging the HFI layer to match the geotransform, projection and dimensions of the tpi layer?
Yeah that's the direction I received from Neal, unless I misconstrued it but perhaps we should revisit that.

Reply:
Ya I didn't think about projections but I guess what we're doing with the HFI is going from lambert (original) to WGS84 (pmtiles) then re-projecting again back to the DEM crs. Probably not ideal.
Curious to see how the HFI data looks at a pixel level after all this when compared to an original tif in an area where there's no snow

Reply:
Maybe I should spend the time to adjust our HFI geospatial pipeline to save the snow-masked HFI as rasters as well as pmtimes.

Reply:
Ok going to adjust this based on the work merged here: https://github.com/bcgov/wps/pull/3809